spec/components/search_spec.rb

```diff
@@ -341,12 +341,22 @@ def new_post(raw, topic = nil)
     end
 
     context "search for a topic by url" do
-      let(:result) { Search.execute(topic.relative_url, search_for_id: true, type_filter: 'topic') }
 
       it 'returns the topic' do
+        result = Search.execute(topic.relative_url, search_for_id: true, type_filter: 'topic')
         expect(result.posts.length).to eq(1)
         expect(result.posts.first.id).to eq(post.id)
       end
+
+      context 'restrict_to_archetype' do
+        let(:personal_message) { Fabricate(:private_message_topic) }
+        let!(:p1) { Fabricate(:post, topic: personal_message, post_number: 1) }
+
+        it 'restricts result to specified archetype' do
+          result = Search.execute(personal_message.relative_url, search_for_id: true, type_filter: 'topic', guardian: Guardian.new(Fabricate(:admin)), restrict_to_archetype: 'regular')
```

Comment:
```suggestion
          result = Search.execute(personal_message.relative_url, search_for_id: true, type_filter: 'topic', guardian: Guardian.new(Fabricate(:admin)), restrict_to_archetype: ArcheType.default)
```

Reply:
Made the change in latest commit. Notice `ArcheType` vs `Archetype` :)